synapse/storage/databases/main/roommember.py

```diff
@@ -1147,6 +1169,27 @@ async def _get_joined_hosts(
 
         return frozenset(cache.hosts_to_joined_users)
 
+    # TODO: this _might_ turn out to need caching, let's see
```

Comment:
Can we remove this todo?

Reply:
Sure.

Erik pointed out that we should(?) only hit this function if there was a cache miss for `get_joined_hosts`; an extra layer of caching isn't likely to get us anything.